api/commands/compile.ts

```diff
@@ -10,6 +10,17 @@ export async function compile(
 ): Promise<dataform.CompiledGraph> {
   // Resolve the path in case it hasn't been resolved already.
   path.resolve(compileConfig.projectDir);
+
+  try {
+    // check dataformJson is valid before we try to compile
+    const dataformJson = fs.readFileSync(`${compileConfig.projectDir}/dataform.json`, "utf8");
+    JSON.parse(dataformJson);
+  } catch (e) {
+    const error = new Error("Compile Error: `dataform.json` is invalid");
```

Comment:
this stacktrace is probably a lot cleaner... can we just `throw new Error("Contents of 'dataform.json' file are invalid.");`?